library/htmlspecialchars.inc.php

```diff
@@ -20,6 +20,15 @@ function js_escape($text)
     return json_encode($text);
 }
 
+/**
+ * Escape a javascript literal for maths.
+ */
+function js_escape_num($text)
+{
+    return json_encode($text, JSON_NUMERIC_CHECK);
```

Comment:
This needs to be used carefully. It may work in this case but create hidden issues in others.
If json_encode is passed a true integer then it should encode as an integer.
However, each version of PHP seems to be unpredictable with issues like this.
We ran into this encode issue awhile back I think PHP 5.6 or 7 and determined the encoding was working correctly.
Here I think we believe an integer is being passed but it's a string in reality.

I guess I don't have an issue with this solution but I'm wary it will get misused. You might check out some of the issues this can cause.

Reply:
thanks @sjpadgett for weighing in, sounds like it's not too much trouble to use `parseInt()` when @ruthkonyn is calculating the score and revert to `js_escape()`

Reply:
If Brady is okay with this solution then I'm as well. I am not the best at keeping up with escaping.
Personally I never trust passed in values in javascript and handle accordingly. This goes for Booleans and math.
I tend to test my variables and cast depending on their use case. In the case of javascript I think in terms of strict typing just to stay out of trouble. In terms of PR reviews, seems I can't stay out of trouble!:)

I though the parseInt() would have been the logical solution. But, this works too.

Reply:
thanks you @sjpadgett  - i'd rather not introduce a potential misuse issue - so happy to find another solution. 
what are your thoughts on my using the following to pick up the integer value of the score? 

 $obj = formFetch("form_phq9", $_GET["id"]);

```
  var i = <?php echo text($obj['interest_score']); ?>; //the value from last time
                           phq9_score +=  i;
```

Reply:
thanks all, i'll go with @stephenwaite's js_escape/parseInt solution, i've already tried it out and it does work. I'll resubmit tomorrow. 
thanks for weighing in  @sjpadgett !!- i have found the discussion  very useful.